syntax = "proto3";

package anomalydetection;

option go_package = "./proto";

service AnomalyDetection {
  rpc DetectAnomaly (AnomalyRequest) returns (AnomalyResponse);
  rpc StreamDetectAnomaly (stream AnomalyRequest) returns (stream AnomalyResponse);
}

message AnomalyRequest {
  string timestamp = 1;
  string level = 2;
  string component = 3;
  string message = 4;
  int32 message_length = 5;
  int32 has_error_keywords = 6;
  int32 hour = 7;
  int32 day_of_week = 8;
  int32 is_weekend = 9;
}

message AnomalyResponse {
  double anomaly_score = 1;
  bool is_anomaly = 2;
  string original_log = 3;
  string recommendation = 4;
}